streams/src/test/java/org/apache/kafka/streams/processor/internals/StateConsumerTest.java

```diff
@@ -0,0 +1,165 @@
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.MockConsumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.common.utils.Utils;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+public class StateConsumerTest {
+
+    private static final long FLUSH_INTERVAL = 1000L;
+    private final TopicPartition topicOne = new TopicPartition("topic-one", 1);
+    private final TopicPartition topicTwo = new TopicPartition("topic-two", 1);
+    private final MockTime time = new MockTime();
+    private final MockConsumer<byte[], byte[]> consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+    private final Map<TopicPartition, Long> partitionOffsets = new HashMap<>();
+    private GlobalStreamThread.StateConsumer stateConsumer;
+    private StateMaintainerStub stateMaintainer;
+
+    @Before
+    public void setUp() throws Exception {
+        partitionOffsets.put(topicOne, 20L);
+        partitionOffsets.put(topicTwo, 30L);
+        stateMaintainer = new StateMaintainerStub(partitionOffsets);
+        stateConsumer = new GlobalStreamThread.StateConsumer(consumer, stateMaintainer, time, 10L, FLUSH_INTERVAL);
+    }
+
+    @Test
+    public void shouldAssignPartitionsToConsumer() throws Exception {
+        stateConsumer.initialize();
+        assertEquals(Utils.mkSet(topicOne, topicTwo), consumer.assignment());
+    }
+
+    @Test
+    public void shouldSeekToInitialOffsets() throws Exception {
+        stateConsumer.initialize();
+        assertEquals(20L, consumer.position(topicOne));
+        assertEquals(30L, consumer.position(topicTwo));
+    }
+
+    @Test
+    public void shouldUpdateStateWithReceivedRecordsForPartition() throws Exception {
+        stateConsumer.initialize();
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 20L, new byte[0], new byte[0]));
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 21L, new byte[0], new byte[0]));
+        stateConsumer.pollAndUpdate();
+        assertEquals(2, stateMaintainer.updatedPartitions.get(topicOne).intValue());
+    }
+
+    @Test
+    public void shouldUpdateStateWithReceivedRecordsForAllTopicPartition() throws Exception {
```

Comment:
It is a question of how you write tests. I write tests and code starting from the simplest things and working out from that. So the test for the single topic comes first, and then the test for multiple topics. In this case the test for multiple topics might be enough, but maybe it isn't, maybe the code assume there is always >1 topic? It is good to have both